src/tools/gravity-parseList.c

```diff
@@ -352,14 +365,24 @@ int gravity_parseList(const char *infile, const char *outfile, const char *adlis
 
 			// Skip IP addresses
 			// IPv4 addresses
-			struct in_addr buffer = { 0 };
-			if (inet_pton(AF_INET, token, &buffer) == 1)
-				goto next_domain;
+			// Don't test any token not starting with a digit
+			if (token[0] <= '9' && token[0] >= '0')
+			{
+				// Potential IPv4 address
+				struct in_addr buffer = { 0 };
+				if (inet_pton(AF_INET, token, &buffer) == 1)
+					goto next_domain;
+			}
 
-			// IPv6 addresses
-			struct in6_addr buffer6 = { 0 };
-			if (inet_pton(AF_INET6, token, &buffer6) == 1)
-				goto next_domain;
+			// Ipv6 address
+			// Only test for IPv6 address if line contained a colon
+			// and it's in this token's first 5 characters
+			if (line_has_colon && memchr(token, ':', 5) != NULL)
```

Comment:
```suggestion
			if (strchr(line, ':') != NULL && memchr(token, ':', 5) != NULL)
```
Avoid `strchr()` - which walks the entire line - to be called for IPv4 address lines at all.

Reply:
I'm also wondering it this may be a possible place for the code to crash. `memchr` accesses up to 5 bytes inside `token` but couldn't the latter be shorter than that? So `memchr` possibly looks into uninitialized/unallocated memory?

Something like
``` c
memchr(token, ':', min(strlen(token), 5))
```
would be safer, however, `strlen()` alone will already walk the entire line so you could even use a tight loop here:
``` c
static inline bool string_has_within(const char *s, const char character, const int maxlen)
{
    for (int i = 0; i < maxlen && s[i] != '\0'; ++i)
        if (s[i] == character)
            return true;
    return false;
}
```
and then
```suggestion
			if (string_has_within(token, ':', 5))
```
(note: not tested)